### PR TITLE
Philips HUE lamps Proxy Minion

### PR DESCRIPTION
**Pull request by isbm**
During our [SUSECon 2015](http://www.susecon.com/) in Amsterdam this November, we will be performing various SaltStack and SUSE Manager demonstrations. One of those demos will be controlling a set of [Philips HUE lamps](http://www2.meethue.com/).

This is a _really_ working Proxy Minion. :wink: 

Setup in pillars:

```
top.sls

base:
  'philips':
    - philips


philips.sls

proxy:
  proxytype: philips_hue
  host: [hostname or IP address]
  user: [username]
```

Example usage:

To see all information about available lamps:

```
salt philips hue.lights
```

To switch (toggle) their status (those ON goes OFF and vice versa):

```
salt philips hue.switch
```

Or bring specific lamp to a specific status:

```
salt philips hue.switch id=1 on=true
```

It supports color setting (presets colors or your own custom, gamut/XY based), alerting, color wheel effect, configurable blinking, lamp naming/renaming, brightness and temperature. Now you can make your NOC room glowing red and blinking, when your servers goes down. :wink:

### CODE REVIEW

File: salt/proxy/philips_hue.py [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+
+'''
+Philips HUE lamps module for proxy.
+'''
+
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+import requests
```
Comment (s0undt3ch) [inline salt/proxy/philips_hue.py]: It would be awesome if this could use salt.utils.http instead of requests directly....
Just an idea...

**Comment by s0undt3ch** [comment]
Got to get me some of these bulbs now....

:+1:

File: salt/proxy/philips_hue.py (hunk 1 above)
Comment (isbm) [inline salt/proxy/philips_hue.py]: @s0undt3ch I will look at that later, in the separate PR.

**Comment by rallytime** [comment]
@cro Can you review this one, please?

**Comment by isbm** [comment]
@rallytime @cro If Jenkins will quack, I will fix the issues later.

**Comment by cro** [comment]
@isbm I am totally impressed.  This is really, really neat!!

As a heads up, putting the proxymodule in `__opts__` will be deprecated come 2015.8.2.  In it's place, we are exposing a global dunder variable `__proxy__` (like `__salt__` or `__grains__`).  There's a transition path via a config variable called `add_proxymodule_to_opts` which will default to True in 2015.8.2 and flip to False in Boron.

Looks like there's just a little lint in philips_hue.py, and I'll watch for your switch to salt.utils.http in a separate PR like you said.

File: salt/proxy/philips_hue.py (hunk 1 above)
Comment (s0undt3ch) [inline salt/proxy/philips_hue.py]: Ok. You'll need to try/except that import for now since requests is not a hard dependency.

**Comment by isbm** [comment]
@s0undt3ch I've try/except'ed the `requests` at the very moment. Later on I will move to `salt.utils.http`.

**Comment by isbm** [comment]
@cro Thanks for the insight, because `__opt__` is a little bit indeed "hacky" :wink: 

Currently I try/except'ed `requests` as a soft dependency and if this gets merged, in the next PR I will move this to `salt.utils.http`, also lintfixes.

**Pull request merged**